indexer/ai/src/main/java/com/google/cloudsearch/ai/AISkillSet.java

```diff
@@ -0,0 +1,107 @@
+package com.google.cloudsearch.ai;
+
+import com.google.cloudsearch.exceptions.InvalidConfigException;
+import org.apache.log4j.BasicConfigurator;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+
+/**
+ * Generates a set of skills based on the configuration.
+ */
+public class AISkillSet {
+
+    private List<AISkill> skills;
+    private Logger log = Logger.getLogger(AISkillSet.class);
+
+    /**
+     * Constructor
+     * @param skillSet      The aiSkillSet object from the Configuration.
+     * @param schema        CloudSearch Schema
+     * @throws InvalidConfigException   Throws exception if the config in invalid.
+     */
+    public AISkillSet(JSONObject skillSet, JSONObject schema) throws InvalidConfigException {
+        BasicConfigurator.configure();
+        this.parse(skillSet, schema);
+    }
+
+    /**
+     * Set the skill set.
+     * @param skills  List of AI Skills
+     */
+    public void setSkills(List<AISkill> skills) {
+        this.skills = skills;
+    }
+
+    /**
+     * Returns the set of skills.
+     * @return
+     */
+    public List<AISkill> getSkills() {
+        return this.skills;
+    }
+
+    /**
+     * Parse the skill configuration and instantiate the AISkill objects.
+     * @param skillSet  The aiSkillSet object from the Configuration.
+     * @param schema    CloudSearch Schema
+     * @throws InvalidConfigException   Throws exception if the config is invalid.
+     */
+    public void parse(JSONObject skillSet, JSONObject schema) throws InvalidConfigException {
+        List<AISkill> skillList = new ArrayList();
+        JSONArray skillsArray = (JSONArray) skillSet.get(Constants.CONFIG_SKILL_SET);
+        if(skillsArray == null) {
+            throw new InvalidConfigException("AI Skill Set not specified in configuration.");
+        }
+        log.info("Skills specified :");
+        for(Object skillObj : skillsArray) {
+            JSONObject currentSkill = (JSONObject) skillObj;
+            String skillName;
+            String[] nameParts;
+            skillName = (String) currentSkill.get(Constants.CONFIG_SKILL_NAME);
+            if(skillName == null || skillName.equals("")) {
```

Comment:
nit: use the empty check instead of comparing to empty string

Reply:
done